git/remote.py

```diff
@@ -694,10 +715,26 @@ def push(self, refspec=None, progress=None, **kwargs):
 
         :param refspec: see 'fetch' method
         :param progress:
-            Instance of type RemoteProgress allowing the caller to receive
-            progress information until the method returns.
             If None, progress information will be discarded
 
+            No further progress information is returned after push returns.
+
+            A function (callable) that is called with the progress infomation:
+
+            progress( op_code, cur_count, max_count=None, message='' )
+
+            op_code is a bit mask of values defined in git.RemoteProgress
+
+            cur_count and max_count are float values.
+
+            max_count is None if there is no max_count
+
+            messages is '' if there is no additon message.
+
+            Deprecated: Pass in a class derived from git.RemoteProgres that
```

Comment:
I see no reason to deprecate this - it should be fine to just remove `Deprecated: ` here.